src/lib/utils/ct_utils.h

```diff
@@ -64,15 +67,89 @@ constexpr inline void unpoison(const T* p, size_t n) {
    BOTAN_UNUSED(p, n);
 }
 
+//
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+//
+// Convenience overloads for poisoning/unpoisoning various types and ranges
+//
+
+template <std::integral T>
+constexpr void poison(T& p) {
+   poison(&p, sizeof(T));
+}
+
+template <std::integral T>
+constexpr void unpoison(T& p) {
+   unpoison(&p, sizeof(T));
+}
+
+template <ranges::spanable_range R>
+constexpr void poison(R&& r) {
+   std::span s{r};
+   poison(s.data(), s.size());
+}
+
+template <ranges::spanable_range R>
+constexpr void unpoison(R&& r) {
+   std::span s{r};
+   unpoison(s.data(), s.size());
+}
+
+/**
+ * Poison a class type that provides a public `const_time_poison()` method
+ * For instance: BigInt, CT::Mask<>, FrodoMatrix, ...
+ */
 template <typename T>
-constexpr inline void unpoison(T& p) {
-#if defined(BOTAN_HAS_VALGRIND)
-   if(!std::is_constant_evaluated()) {
-      VALGRIND_MAKE_MEM_DEFINED(&p, sizeof(T));
-   }
-#endif
+   requires requires(const T& x) { x.const_time_poison(); }
+constexpr void poison(const T& x) {
+   x.const_time_poison();
+}
+
+/**
+ * Unpoison a class type that provides a public `const_time_unpoison()` method
+ * For instance: BigInt, CT::Mask<>, FrodoMatrix, ...
+ */
+template <typename T>
+   requires requires(const T& x) { x.const_time_unpoison(); }
+constexpr void unpoison(const T& x) {
+   x.const_time_unpoison();
+}
```

Comment:
We saw that `BitInt::const_time_poison()` was a thing already and just extended the idea to general types. This will be convenient for the PQC stuff in particular, as it often has helper-structures that could encapsulate their poisoning.

Ideally, this method should be `_const_time_poison()`, I guess, but we didn't want to break the existing `BigInt` API.

Reply:
We could alternately name it `_const_time_poison` and on `BigInt` just define `const_time_poison` as a deprecated forward.

*Or* just define another instance and handle both cases --

```
template <typename T>
   requires requires(const T& x) { x._const_time_unpoison(); }
constexpr void unpoison(const T& x) {
   x._const_time_unpoison();
}
```

Reply:
I'll rename them to `_....` which is anyway the new convention.